src/api/java/Context.java

```diff
@@ -723,7 +723,7 @@ public BoolExpr mkBool(boolean value)
     /**
      * Creates the equality {@code x = y}
      **/
-    public <R extends Sort> BoolExpr mkEq(Expr<R> x, Expr<R> y)
+    public BoolExpr mkEq(Expr<?> x, Expr<?> y)
```

Comment:
Now the issue is inverted; this will accept arbitrary exprs for comparison (e.g. a sequence and an integral sort). I don't see a good solution for fixing the mixed arith without adding a new method, though...

Reply:
Indeed it is possible to compare expr that cannot be equal because of their sort. 
But I think it is better to allow this rather than not begin able to express valid comparisons.

I totally agree with you that this is suboptimal, but essentially this is a limitation of Java.

Reply:
I would support the new version because equality has some overloading properties that cannot be captured in the type system. At least Z3 throws a runtime exception.